utils/measure/runner/charging.py

```diff
@@ -43,8 +43,29 @@ def run(
         )
         input("Hit enter when you are ready to start..")
 
+        print()
+
         battery_level = self.controller.get_battery_level()
         measurements: dict[int, list[float]] = {}
+        is_charging = self.controller.is_charging()
+        is_valid_state = self.controller.is_valid_state()
```

Comment:
Please extract all the logic to a new method `wait_for_vacuum_to_start_charging`